cloud/kyup/kyup.py

```diff
@@ -0,0 +1,338 @@
+    for i in opt:
+        opt[i] =  module.params.get(i)
+        if opt[i] is None:
+            module.fail_json(changed=False, msg = '%s parameter is required for creating a container' % i)
+
+    # check if the container already exists
+    req = '{"action":"cloudList","authorization_key":"%s","data":{}}'
+    ret = api_request(module, req % module.params.get('api_key'))
+    for container in ret['data']['list']:
+        if container['name'] == module.params.get('name'):
+            container['ip'] = container['ip'][0:-3]
+            module.exit_json(
+                changed=False,
+                ansible_facts = { 'container_name': container['name'], 'container_ip': container['ip'] },
+                msg = 'Container %s already exists' % module.params.get('name')
+            )
+
+    storage_type = module.params['storage_type']
+    if storage_type is None or storage_type == 'local':
+        storage_type = 0
+    elif storage_type == 'distributed':
+        storage_type = 1
+
+    container_id = 0
+    req = '{"action":"cloudCreate","authorization_key":"%s","data":{"name":"%s","password":"%s","image_name":"%s","datacenter_id":%d,"storage_type":%d,"resources":{"mem":%d,"hdd":%d,"cpu":%d,"bw":%d}}}'
+    ret = api_request(module, req % (
+            module.params.get('api_key'),
+            opt['name'],
+            encrypt(enc_key, opt['password']),
+            opt['image'],
+            opt['dc_id'],
+            storage_type,
+            module.params.get('mem'),
+            module.params.get('hdd'),
+            module.params.get('cpu_cores'),
+            module.params.get('bw')
+            )
+        )
+    if 'task_id' in ret['data']:
+        if ret['data']['task_id'] > 0:
+            container_id = get_task_status(module, ret['data']['task_id'])
+            container = kyup_action(module, 'cloudDetails', container_id)
+            add_ssh_keys(module, container_id)
+            # remove the mask portion of the string
+            container['ip'] = container['ip'][0:-3]
+            module.exit_json(
+                changed = True,
+                ansible_facts = { 'container_name': container['name'], 'container_ip': container['ip'] },
+                msg = 'Container: Name: %s IP: %s' % (container['name'], container['ip'])
+            )
+        else:
+            module.fail_json(changed=True, msg = 'Invalid task_id')
+    else:
+        module.fail_json(changed=True, msg = 'Unable to get task_id')
+
+def core(module):
+    api_key = module.params['api_key'] or os.environ['KYUP_API_KEY']
+    if api_key is None:
+        module.fail_json(changed=False, msg = 'you can not continue without api_key')
+    module.params['api_key'] = api_key
+
+    action = module.params['action']
+    if action is None:
+        module.fail_json(changed=False, msg = 'action parameter is required by this module')
+    elif action == 'create':
+        create_container(module)
+    else:
+        if action == 'destroy':
```

Comment:
You can create a class with static method for each action and call them with getattr, this might simplify the extensibility for the `action` argument. for example

```python
class Actions:
    @staticmethod
    def create(module):
        return create_container(module)

action = getattr(Actions, action)
action(module)
```